server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -436,7 +436,8 @@ protected final void closeNoLock(String reason, CountDownLatch closedLatch) {
                  This is not required for remote store implementations given on failover the replica re-syncs with the store
                  during promotion.
                  */
-                if (engineConfig.getIndexSettings().isRemoteStoreEnabled() == false) {
+                if (engineConfig.getIndexSettings().isRemoteStoreEnabled() == false
+                    && engineConfig.getIndexSettings().isRemoteNode() == false) {
```

Comment:
Do we need both the conditions or having `engineConfig.getIndexSettings().isRemoteNode() == false` is sufficient?

Reply:
Yes, it would be possible when we support remote store to doc rep migration .